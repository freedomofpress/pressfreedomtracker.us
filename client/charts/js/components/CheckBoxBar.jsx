import React from 'react'

const RADIO_BOX_WIDTH = 35

export default function CheckBoxBar({ label, count, i, onClick, isSelected, barWidth, width }) {
	return (
		<div
			style={{
				display: 'flex',
				flexDirection: 'row',
				marginTop: 12,
				marginBottom: 12,
			}}
		>
			<div
				style={{
					display: 'flex',
					flexDirection: 'row-reverse',
					width: RADIO_BOX_WIDTH,
				}}
			>
				<input
					type="checkbox"
					name="drone"
					checked={isSelected}
					onChange={onClick}
					style={{ width: 24, height: 24 }}
					disabled={count < 1}
				/>
			</div>

			<div
				style={{
					display: 'flex',
					flexDirection: 'column',
					justifyContent: 'flex-end',
				}}
			>
				<div
					style={{
						display: 'flex',
						flexDirection: 'row',
						justifyContent: 'space-between',
						marginBottom: 0,
					}}
				>
					<div
						style={{
							color: 'black',
							fontSize: 14,
							lineHeight: '20px',
							fontFamily: 'var(--font-base)',
						}}
					>
						{label}
					</div>
					<div
						style={{
							fontFamily: 'var(--font-mono)',
							fontSize: 12,
							justifyContent: 'flex-end',
							backgroundColor: isSelected ? '#F2FC67' : null,
							padding: 3,
						}}
					>
						{count}
					</div>
				</div>
				<div style={{ width: width - RADIO_BOX_WIDTH, height: 4 }}>
					<div
						style={{
							zIndex: -1,
							width: barWidth,
							height: 3,
							border: '1px solid black',
							backgroundColor: 'black',
							transition: 'all 250ms ',
						}}
					/>
					<div
						style={{
							top: -4,
							left: 1,
							position: 'relative',
							width: isSelected ? barWidth : 0,
							height: 3,
							backgroundColor: '#F2FC67',
							transition: 'all 250ms ',
						}}
					/>
				</div>
			</div>
		</div>
	)
}
