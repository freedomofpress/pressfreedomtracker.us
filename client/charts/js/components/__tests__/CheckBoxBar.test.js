import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import CheckBoxBar from '../CheckBoxBar'

test('renders CheckBoxBar with mocked data', () => {
	expect(ReactTestRenderer.create(
		<CheckBoxBar
			label="test"
			count={2}
			isSelected={false}
			onClick={() => {}}
		/>
	)).toMatchSnapshot();
});
