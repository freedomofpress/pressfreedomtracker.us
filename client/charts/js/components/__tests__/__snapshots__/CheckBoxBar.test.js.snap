// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CheckBoxBar with mocked data 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "marginBottom": 12,
      "marginTop": 12,
    }
  }
>
  <div
    style={
      {
        "display": "flex",
        "flexDirection": "row-reverse",
        "width": 35,
      }
    }
  >
    <input
      checked={false}
      disabled={false}
      name="drone"
      onChange={[Function]}
      style={
        {
          "height": 24,
          "width": 24,
        }
      }
      type="checkbox"
    />
  </div>
  <div
    style={
      {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "flex-end",
      }
    }
  >
    <div
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 0,
        }
      }
    >
      <div
        style={
          {
            "color": "black",
            "fontFamily": "var(--font-base)",
            "fontSize": 14,
            "lineHeight": "20px",
          }
        }
      >
        test
      </div>
      <div
        style={
          {
            "backgroundColor": null,
            "fontFamily": "var(--font-mono)",
            "fontSize": 12,
            "justifyContent": "flex-end",
            "padding": 3,
          }
        }
      >
        2
      </div>
    </div>
    <div
      style={
        {
          "height": 4,
          "width": NaN,
        }
      }
    >
      <div
        style={
          {
            "backgroundColor": "black",
            "border": "1px solid black",
            "height": 3,
            "transition": "all 250ms ",
            "width": undefined,
            "zIndex": -1,
          }
        }
      />
      <div
        style={
          {
            "backgroundColor": "#F2FC67",
            "height": 3,
            "left": 1,
            "position": "relative",
            "top": -4,
            "transition": "all 250ms ",
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
